ui-manchette-with-spacetimechart/src/hooks/useManchetteWithSpaceTimeChart.ts

```diff
@@ -0,0 +1,235 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import type {
+  ProjectPathTrainResult,
+  OperationalPointType,
+} from '@osrd-project/ui-manchette/src/types';
+import type {
+  OperationalPoint,
+  SpaceScale,
+  SpaceTimeChartProps,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import { useIsOverflow } from './useIsOverFlow';
+import usePaths from './usePaths';
+import { MAX_ZOOM_Y, MIN_ZOOM_Y, ZOOM_Y_DELTA, INITIAL_SPACE_TIME_CHART_HEIGHT } from '../consts';
+import {
+  calcOperationalPointsToDisplay,
+  computeTimeWindow,
+  getOperationalPointsWithPosition,
+  getScales,
+  calcOperationalPointsHeight,
+  zoomX,
+} from '../helpers';
+import { getDiff } from '../utils/point';
+
+type State = {
+  xZoom: number;
+  yZoom: number;
+  xOffset: number;
+  yOffset: number;
+  panning: { initialOffset: { x: number; y: number } } | null;
+  scrollPosition: number;
+  isProportional: boolean;
+  operationalPointsChart: OperationalPoint[];
+  panY: boolean;
+  scales: SpaceScale[];
+};
+
+const useManchettesWithSpaceTimeChart = (
+  operationalPoints: OperationalPointType[],
+  projectPathTrainResult: ProjectPathTrainResult[],
+  manchette: React.RefObject<HTMLDivElement>,
```

Comment:
Can you also rename this props ?
```suggestion
  manchetteWithSpaceTimeChartContainer: React.RefObject<HTMLDivElement>,
```